jujugui/static/gui/src/app/init.js

```diff
@@ -333,11 +333,45 @@ class GUIApp {
     this.topology = this._renderTopology();
 
     /**
-      Once login is complete this will hold the connection to the controller
+      Once login is complete this will hold the connection to the controller.
       @type {Object}
     */
     this.controllerConnection = null;
 
+    /**
+      Once login is complete this will hold the connection to the model.
+      @type {Object}
+    */
+    this.modelConnection = null;
+
+    /**
+      The model uuid that is currently connected to.
+      @type {string}
+    */
+    this.activeModelUUID = null;
+
+    /**
+      Stores the status of the model connection so we can intellegently
+      determine what to do with the modelConnection. This is required even
+      though the connection is asynchronous because we want to block dispatching
+      the application until the model connection is completely ready and
+      native promises do not allow external inspection on its status.
+
+      Allowed values are:
+        null
+        'ready'
+        'connecting'
+      @type {String}
+    */
+    this.modelConnectionStatus = null;
```

Comment:
Should we use a getter/setter for this to enforce the allowed values?

Reply:
good call

Reply:
done